httpclient5-cache/src/test/java/org/apache/hc/client5/http/impl/cache/TestProtocolRequirements.java

```diff
@@ -2123,7 +2123,13 @@ public void testAgeHeaderPopulatedFromCacheEntryCurrentAge() throws Exception {
         final ClassicHttpResponse result = execute(request);
 
         Assertions.assertEquals(200, result.getCode());
-        Assertions.assertEquals("11", result.getFirstHeader("Age").getValue());
+        // We calculate the age of the cache entry as per RFC 9111:
+        // We first find the "corrected_initial_age" which is the maximum of "apparentAge" and "correctedReceivedAge".
+        // In this case, max(1, 2) = 2 seconds.
+        // We then add the "residentTime" which is "now - responseTime",
+        // which is the current time minus the time the cache entry was created. In this case, that is 8 seconds.
+        // So, the total age is "corrected_initial_age" + "residentTime" = 2 + 8 = 10 seconds.
+        Assertions.assertEquals("10", result.getFirstHeader("Age").getValue());
```

Comment:
@arturobernalg  I find `ContainsHeaderMatcher` look a bit nicer here .

Reply:
@ok2c 
The current implementation of ContainsHeaderMatcher does not directly support the object types we're working with here, such as ClassicHttpResponse. Adapting it to these types may require significant changes to the existing code. Maybe I'm missing something.

Reply:
@arturobernalg Ah, i fixed the matcher in my branch but not in 5.3.x. Please pull the latest changes.

Reply:
@ok2c  Change applied.